test/test_selectors.py

```diff
@@ -613,6 +612,34 @@ def test_leaking_fds(self):
         after_fds = len(proc.open_files())
         self.assertEqual(before_fds, after_fds)
 
+    @skipIf(sys.platform == "win32", "psutil.Process.open_files() is unstable on Windows.")
+    def test_wait_read_leaking_fds(self):
+        old_selector = wait.DefaultSelector
+        wait.DefaultSelector = self.SELECTOR
+        self.addCleanup(setattr, wait, "DefaultSelector", old_selector)
+
+        proc = psutil.Process()
+        rd, wr = self.make_socketpair()
+        before_fds = len(proc.open_files())
+        self.assertEqual(0, len(wait.wait_for_read([rd], 0.001)))
+        after_fds = len(proc.open_files())
+        self.assertEqual(before_fds, after_fds)
+
+    @skipIf(sys.platform == "win32", "psutil.Process.open_files() is unstable on Windows.")
+    def test_wait_write_leaking_fds(self):
+        old_selector = wait.DefaultSelector
+        wait.DefaultSelector = self.SELECTOR
+        self.addCleanup(setattr, wait, "DefaultSelector", old_selector)
+
+        proc = psutil.Process()
+        rd, wr = self.make_socketpair()
+        before_fds = len(proc.open_files())
+        self.assertEqual(1, len(wait.wait_for_write([wr], 0.001)))
+        after_fds = len(proc.open_files())
+        self.assertEqual(before_fds, after_fds)
```

Comment:
I can add a test for that as well, I figured check the 'bad outcome' we're trying to avoid. :)